### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by rachidelaid**
> tasks done :heavy_check_mark:  :seedling: :
- created the `factorial` method in the class `Solver` using TDD. 
- created the `reverse` method in the class `Solver` using TDD.
- created the `fizzbuzz` method in the class `Solver` using TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  context 'test the factorial method' do
+    it 'factorial 4 should return 24' do
+      num = 4
+      result = @solver.factorial(num)
+      expect(result).to eq(24)
+    end
+  end
+
```
Comment (Dipeshtwis) [inline spec/solver_spec.rb]: - [ ] I would recommend adding more tests when `num = 0` the expected result is `1` or when the value is negative, it would raise an exception.

 Currently, I tried these tests but the tests are failing. Please update the method as well accordingly.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @rachidelaid & @Ginohmk, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
:+1: Followed the TDD approach.
:+1: README is descriptive.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ahmetbozaci — APPROVED** [review]
# STATUS: APPROVED 🥳 🥳 🥳 



@rachidelaid @Ginohmk   ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

aproved

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_
